# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/blindcoder/kterm/kterm-3dsc-linux.patch
# ROCK Linux is Copyright (C) 1998 - 2003 Clifford Wolf
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version. A copy of the GNU General Public
# License can be found at Documentation/COPYING.
# 
# Many people helped and are helping developing ROCK Linux. Please
# have a look at http://www.rocklinux.org/ and the Documentation/TEAM
# file for details.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -uNr KTerm.ad KTerm.ad
--- ./KTerm.ad	Sun Jun 23 10:00:22 1996
+++ ./KTerm.ad	Sat May  4 10:06:44 2002
@@ -123,3 +123,6 @@
 *tek4014*font3: 6x13
 *tek4014*fontSmall: 6x10
 
+*scrollBar:			true
+*Scrollbar.background:		gray
+
diff -uNr kterm.h kterm.h
--- ./kterm.h	Fri Jul 12 07:01:37 1996
+++ ./kterm.h	Sat May  4 10:06:44 2002
@@ -36,6 +36,6 @@
 #define KTERM_KINPUT2	/* Kinput2 protocol */
 #define KTERM_COLOR	/* color sequence */
 #define KTERM_NOTEK	/* disables Tektronix emulation */
-#undef  KTERM_XAW3D	/* Xaw3d -DARROW_SCROLLBAR support */
+#define KTERM_XAW3D	/* Xaw3d -DARROW_SCROLLBAR support */
 
 #endif /* !_KTERM_H_ */
diff -uNr main.c main.c
--- ./main.c	Fri Jul 12 07:01:38 1996
+++ ./main.c	Sat May  4 10:06:44 2002
@@ -140,6 +140,7 @@
 #define HAS_UTMP_UT_HOST
 #define LASTLOG
 #define WTMP
+#include <termios.h>
 #endif
 
 #include <sys/ioctl.h>
diff -uNr screen.c screen.c
--- ./screen.c	Sun Jun 23 10:00:26 1996
+++ ./screen.c	Sat May  4 10:06:44 2002
@@ -46,8 +46,9 @@
 #include <sys/ptem.h>
 #endif
 
-extern Char *calloc(), *malloc(), *realloc();
-extern void free();
+#include <stdlib.h>
+/* extern Char *calloc(), *malloc(), *realloc();
+   extern void free(); */
 
 ScrnBuf Allocate (nrow, ncol, addr)
 /*
diff -uNr scrollbar.c scrollbar.c
--- ./scrollbar.c	Fri Jul 12 07:01:39 1996
+++ ./scrollbar.c	Sat May  4 10:06:44 2002
@@ -29,6 +29,7 @@
 #include "ptyx.h"		/* gets Xt headers, too */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <ctype.h>
 #include <X11/Xatom.h>
 
@@ -324,7 +325,7 @@
 	register TScreen *screen = &xw->screen;
 	register int border = 2 * screen->border;
 	register int i;
-	Char *realloc(), *calloc();
+	/* Char *realloc(), *calloc(); */
 
 	if(screen->scrollbar)
 		return;
